expyfun/stimuli/_tracker.py

```diff
@@ -239,6 +248,22 @@ def respond(self, correct):
                 if self._direction <= 0:
                     self._direction = 1
 
+        if self._repeat_limit == 'reversals':
+            if self._x[-1] == self._x_min:
```

Comment:
Right, yeah that was the issue. If you count a reversal but don't change direction then it creates weird situations where you can be moving down, and then eventually be moving up after an even number of reversals, which shouldn't be possible.

Without the limit, the subject wills stay at the limit until they eventually get a wrong answer and a reversal, moving up. The limit essentially fast-forwards to that point, skipping a bunch of useless correct answers.